### PR TITLE
API: Data Access Object refacto

### LINKED ISSUE #348: API: refactor db accessor and 4xx error msg

### PR DESCRIPTION
**Pull request by HugoGresse**
New Data Access Object for reminder/user/room

Close #348

### CODE REVIEW

File: functions/src/api/v1/reminders/editReminder.ts [hunk 1]
```diff
@@ -99,9 +95,9 @@ export const editReminder = wrap(async (req: Request, res: Response) => {
         dataToEdit.hostName = hostName
     }
 
-    await updateReminderDb(dataToEdit)
+    await ReminderDao.update(dataToEdit)
 
-    const reminder = await getReminderDb(reminderId)
+    const reminder = await ReminderDao.get(reminderId)
 
     res.send({
```
Comment (ntchou) [inline functions/src/api/v1/reminders/editReminder.ts:102]: it's not consistent with swagger status code definition. https://github.com/Instant-Visio/InstantVisio-WebApp/blob/1a7d974bab4ef0366cd884265b29fadfc5ad7b17/functions/src/api/v1/reminders/editReminder.ts#L44 You should change swagger definition to `200`

File: functions/src/db/UserDao.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        if (!userDocumentSnapshot?.exists) {
+            throw new NotFoundError('Resource does not exist')
+        }
+        return <User>{
+            id: userId,
+            ...userDocumentSnapshot.data(),
+        }
+    }
+
+    public static async update(userId: UID, data: object) {
```
Comment (ntchou) [inline functions/src/db/UserDao.ts:26]: missing return type

**Review by ntchou — COMMENTED** [review]
Some missing return type but good job

File: functions/src/db/RoomDao.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        await db.collection(COLLECTIONS.rooms).doc(roomId).set({
+            uid: userId,
+            password: password,
+            createdAt: serverTimestamp(),
+            updatedAt: serverTimestamp(),
+            service: DEFAULT_ROOM_TYPE,
+            startAt,
+        })
+
+        return roomId
```
[the remark this answers is not in the document]
Comment (HugoGresse) [reply functions/src/db/RoomDao.ts:83]: set is used at the same time a `add` 

```

    if (specificRoomId) {
        roomId = await RoomDao.set(
            userId,
            specificRoomId,
            roomPassword,
            roomStartAt
        )
    } else {
        roomId = await RoomDao.add(userId, roomPassword, roomStartAt)
    }

```

so it convenient to have the same structure.

**Review by MattiaPrimavera — COMMENTED** [review]
Nice job ;)

**Pull request merged**